### PR TITLE
Replace Faraday::Error::* usage

### PR DESCRIPTION
**Pull request by jasonkim**
- Replace `Faraday::Error::*` with `Faraday::*` error classes (following the [upgrade guide](https://github.com/lostisland/faraday/blob/main/UPGRADING.md#errors))
- Handle all `Faraday::Error` instead of `Faraday::Error::ClientError`
- Unlock faraday version to allow 1.0.0
- Bumped up the faraday and faraday middleware gems slightly

### CODE REVIEW

**Review by tomthorogood — COMMENTED** [review]
Although this _looks_ just fine to me, I've only mucked around here once or twice, and feel as though @ndonewar, @richa-d, or @kag728 are in a better place to know whether this could have any undesired side effects. 

Just wanted to let you know I looked, at least! 😅

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 6.0.4 (2024-04-05)
```
Comment (ndonewar) [inline CHANGELOG.md:1]:
```suggestion
## 6.1.0 (2024-04-09)
```

File: lib/elastomer_client/version.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 module ElastomerClient
-  VERSION = "6.0.3"
+  VERSION = "6.0.4"
```
Comment (ndonewar) [inline lib/elastomer_client/version.rb:4]:
```suggestion
  VERSION = "6.1.0"
```

**Review by ndonewar — COMMENTED** [review]
Two minor suggestions about bumping the minor version, but otherwise looks good!

**Review by ndonewar — APPROVED** [review]

**Pull request merged**